lib/models/auth.dart

```diff
@@ -148,8 +148,10 @@ class AuthModel with ChangeNotifier {
       if (info['message'] != null) {
         throw info['message'];
       }
+      if (info['username'] == null) {
+        throw "username returned is null. Make sure token is not being re-used";
+      }
```

Comment:
What is the response in this case? The `info` variable

Reply:
According to the GitLab documentation, seems all errors would be represented in the `message` field.

Reply:
`message` field is null. So are all other fields of `GitlabUser`. I verified it once again.

Reply:
So does it mean the response json is empty, like `{}`?

Reply:
Sorry should've checked the response instead of the variable after serialising it. Here's the response: 

`{error: invalid_token, error_description: Token is expired. You can either do re-authorization or token refresh.}`

I could've just used the `error` field. Might cover some other cases as well.

Reply:
OK. I guess it is caused by legacy version of GitLab...